File: js/palindrome.js
exports.palindrome = function(word) {
    //checking to see if the word is truly a string and if so taking to lowercace, split to array, filter out anything not alpha character then joining
    //then doing the exact same but reversing the array before joining
    //finally returning the boolean for comparing the two 
    if (typeof(word) === 'string') {
        return word.toLowerCase().split('').filter(element => element.match(/[a-z]/)).join()
        === word.toLowerCase().split('').filter(element => element.match(/[a-z]/)).reverse().join();
    //if word type is number I am taking to string so I can split to array, reverse and then join and then using parse int to take back to an int
    //finally returning the boolean for comparing the two
    } else if (typeof(word) === 'number') {
        return word  === parseInt(word.toString().split('').reverse().join(''));
    }
    
};


// palindrome('racecar') //=== true);
// palindrome('Noon') //=== true);
// palindrome('ciVic') //=== true);
// palindrome('nice') //=== false);
// palindrome(434) //=== true);
//palindrome(123) //=== false);

// console.log("The following should be true if you're trying to do the extra portion of this challenge");
// palindrome("Sore was I ere I saw Eros.") //=== true);
// palindrome("A man, a plan, a canal -- Panama") //=== true);

File: python/palindrome.py
import re

def palindrome(word):
    #check to see if input is a number
    if str(word).isnumeric():
        #if number, cast to string, reversing, cast to list then joining and cast back to int
        b_word = int(''.join(list(reversed(str(word)))))
        #return boolean for comparison of input vs reveresd b-word
        return word == b_word
    #else if not a number take word to lower, removing everything but alpha then joining back to word
    #doing same for b_word but also reversing it
    else:
        a_word = ''.join(list(re.sub("[^a-z0-9]+", '', word.lower())))
        b_word = ''.join(list(reversed(re.sub("[^a-z0-9]+", '', word.lower()))))
        #returning the boolean for comparing a vs b
        return a_word == b_word


# print(palindrome('racecar')) #=== true);
# palindrome('Noon') #=== true);
# print(palindrome('ciVic')) #=== true);
# print(palindrome('nice')) #=== false);
# print(palindrome(434)) #=== true);
# print(palindrome(123)) #=== false);

# //console.log("The following should be true if you're trying to do the extra portion of this challenge");
# print(palindrome("Sore was I ere I saw Eros.")) #=== true);
# print(palindrome("A man, a plan, a canal -- Panama")) #=== true);